share/completions/command.fish

```diff
@@ -1,5 +1,5 @@
-complete -c command -n 'test (count (commandline -opc)) -eq 1' -s h -l help -d 'Display help and exit'
-complete -c command -n 'test (count (commandline -opc)) -eq 1' -s a -l all -d 'Print all external commands by the given name'
-complete -c command -n 'test (count (commandline -opc)) -eq 1' -s q -l quiet -l query -d 'Do not print anything, only set exit status'
-complete -c command -n 'test (count (commandline -opc)) -eq 1' -s s -l search -d 'Print the file that would be executed'
+complete -c command -n __fish_is_first_token -s h -l help -d 'Display help and exit'
```

Comment:
I think the commands that do not allow an in-between option should stay the old way. 

For example `command somecmd` never has a valid option in the middle.

Some for `builtin`, `exec` and `for`